mgmtsystem/hawkular.py

```diff
@@ -766,6 +788,372 @@ def delete_resource(self, feed_id, resource_id):
         return r
 
 
+class HawkularInventoryInMetrics(HawkularService):
+    def __init__(self, hostname, port, protocol, auth, tenant_id):
+        """Creates hawkular inventory service instance. For args refer 'HawkularService'"""
+        HawkularService.__init__(self, hostname=hostname, port=port, protocol=protocol,
+                                 auth=auth, tenant_id=tenant_id, entry="hawkular/metrics")
+
+    _stats_available = {
+        'num_server': lambda self: len(self.list_server()),
+        'num_domain': lambda self: len(self.list_domain()),
+        'num_deployment': lambda self: len(self.list_server_deployment()),
+        'num_datasource': lambda self: len(self.list_server_datasource()),
+        'num_messaging': lambda self: len(self.list_messaging()),
+    }
+
+    def list_feed(self):
+        """Returns list of feeds"""
+        entities = []
+        entities_j = self._get('strings/tags/module:inventory,feed:*')
+        if entities_j and entities_j['feed']:
+            for entity_j in entities_j['feed']:
+                entities.append(Feed(entity_j, CanonicalPath('/f;{}'.format(entity_j))))
+        return entities
+
+    def list_server(self, feed_id=None):
+        """Returns list of middleware servers.
+
+          Args:
+            feed_id: Feed id of the resource (optional)
+        """
+        resources = self.list_resource(feed_id=feed_id,
+                                       resource_type_id='WildFly Server',
+                                       include_data=True)
+        resources.extend(self.list_resource(
+            feed_id=feed_id,
+            resource_type_id='Domain WildFly Server',
+            list_children=True,
+            include_data=True))
+        servers = []
+        if resources:
```

Comment:
Same for many of these below, some of which would definitely look better as list comprehensions